### PR TITLE
feat: Align brand color aliases [WEB-215]

### PR DESCRIPTION
**Pull request by jakemhiller**
## Overview

Based on #777 

This aligns our brand colors to the design colors in abstract.

1. Added new `magenta` and `lavender` colors
1. Updated `color-blue-900` to match the design illustration palette
1. Alias brand colors that match existing palette colors
1. Update colors that didn't match a palette color to match (`brand-yellow`, `brand-pink`, `brand-mint`, `brand-darkBlue`)

Two changes are needed in the design brand palette:

1. `brand-mint` should be updated to `color-green-300` which is `#6AEBC0`
2. `brand-orange` should be updated to `color-orange-400` which is `#FF9F3C`

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.13-alpha.0df6e7.0<br/>@codecademy/gamut-styles@3.3.2-alpha.0df6e7.0<br/>@codecademy/gamut@8.2.2-alpha.0df6e7.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ecff0a90da63a023a1bb97f--codecademy-styleguide.netlify.app>https://5ecff0a90da63a023a1bb97f--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5ecff0a90da63a023a1bb97f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.13-alpha.af9319.0<br/>@codecademy/gamut-styles@3.3.2-alpha.af9319.0<br/>@codecademy/gamut@8.2.2-alpha.af9319.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ecff5e22ee1207ff202d269--codecademy-styleguide.netlify.app>https://5ecff5e22ee1207ff202d269--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5ecff5e22ee1207ff202d269>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.13-alpha.e7661b.0<br/>@codecademy/gamut-styles@3.3.2-alpha.e7661b.0<br/>@codecademy/gamut@8.2.2-alpha.e7661b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ecffaac7509080216956a2c--codecademy-styleguide.netlify.app>https://5ecffaac7509080216956a2c--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5ecffaac7509080216956a2c>Deploy Logs</a></p>

File: packages/gamut-styles/utils/variables/index.js [hunk 1]
```diff
@@ -96,14 +96,17 @@ export const colors = {
...
-  yellow: '#ffc740',
-  purple: '#6400e4',
-  pink: '#ff8ef5',
-  mint: '#6df0c2',
-  beige: '#efd9ca',
-  darkBlue: '#141c3a',
+  red: colors.red[500],
+  orange: colors.orange[400],
+  yellow: colors.yellow[500],
+  purple: colors['royalBlue'],
```
Comment (JoshuaKGoldberg) [inline packages/gamut-styles/utils/variables/index.js:102]:
```suggestion
  purple: colors.royalBlue,
```

Comment (jakemhiller) [reply packages/gamut-styles/utils/variables/index.js:102]: intentional, dot notation breaks the type generation

Comment (jakemhiller) [reply packages/gamut-styles/utils/variables/index.js:102]: might be a fun typescript bug, i didn't dig into it very far

Comment (JoshuaKGoldberg) [reply packages/gamut-styles/utils/variables/index.js:102]: Oh what that does look like a TS bug 😨 

```tsx
    const yellow_1: string;
    export { yellow_1 as yellow };
    import purple_1 = royalBlue;
    export { purple_1 as purple };
```

Comment (JoshuaKGoldberg) [reply packages/gamut-styles/utils/variables/index.js:102]: Reducing it to a simple repro:

```js
export const colors = {
  royalBlue: "#6400e4",
};

export const brandColors = {
  purple: colors.royalBlue,
};
```

Comment (JoshuaKGoldberg) [reply packages/gamut-styles/utils/variables/index.js:102]: 👏 @jakemhiller https://github.com/microsoft/TypeScript/issues/38834

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.22-alpha.cc10ef.0<br/>@codecademy/gamut-styles@4.0.1-alpha.cc10ef.0<br/>@codecademy/gamut@9.0.4-alpha.cc10ef.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ee242d962c43c85a77c7a89--codecademy-styleguide.netlify.app>https://5ee242d962c43c85a77c7a89--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5ee242d962c43c85a77c7a89>Deploy Logs</a></p>

**Pull request merged**